comet/archiver.py

```diff
@@ -78,163 +75,151 @@ def __init__(
 
-        # Parse the json data. If anything goes wrong, push the state back into the list. This is someone else's problem now.
         try:
-            state_id = json_data["hash"]
-            time = json_data["time"]
+            id = json_data["hash"]
         except KeyError as key:
-            logger.error("Key {} not found in state data {}.".format(key, json_data))
-            self._pushback("archive_state", data)
-            return
-
-        try:
-            timestamp = datetime.datetime.strptime(time, TIMESTAMP_FORMAT)
-        except ValueError as err:
-            logger.error(
-                "Failure parsing timestamp {} in state data: {}.".format(time, err)
-            )
-            self._pushback("archive_state", data)
-            return
+            logger.error(f"Key {key} not found in data {json_data}")
+            return False
 
-        # Get the state from redis
-        state = self.redis.hget("states", state_id)
-        if state is None:
-            logger.error(
-                "Failure archiving state {}: state is not known to broker/redis.".format(
-                    state_id
-                )
-            )
-            self._pushback("archive_state", data)
-            return
-        state = json.loads(state)
+        if type_ == "archive_state":
+            return db.DatasetState.exists(id)
 
-        state_type = state["type"]
-        db.insert_state(state_id, state_type, timestamp, state)
+        return db.Dataset.from_id(id) is not None
 
-    def _insert_dataset(self, data):
-        logger.info("Archiving dataset {}".format(data))
+    @staticmethod
+    def _load_json(data):
+        """Load and parse a json dataset."""
         json_data = json.loads(data)
 
-        # Parse the json data. If anything goes wrong, push the state back into the list. This is someone else's problem now.
+        # The "dataset" in redis consists of just a hash and a timestamp
         try:
-            ds_id = json_data["hash"]
+            id = json_data["hash"]
             time = json_data["time"]
         except KeyError as key:
-            logger.error("Key {} not found in dataset data {}.".format(key, json_data))
-            self._pushback("archive_dataset", data)
-            return
+            logger.error(f"Key {key} not found in data {json_data}")
+            return None
 
         try:
             timestamp = datetime.datetime.strptime(time, TIMESTAMP_FORMAT)
         except ValueError as err:
+            logger.error(f"Failure parsing timestamp {time}: {err}")
+            return None
+
+        return id, timestamp
+
+    def _insert_state(self, id, timestamp):
+        """Insert a state into the database."""
+
+        item = self.redis.hget("states", id)
+
+        if item is None:
             logger.error(
-                "Failure parsing timestamp {} in dataset data: {}.".format(time, err)
+                f"Failure archiving item {id}. Item is not known to broker/redis."
```

Comment:
```suggestion
                f"Failure archiving state {id}. Item is not known to broker/redis."
```
This message should mention the item type.